walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,177 @@
+            listTransfer.clear();
+        }
+        else{
+            listTransfer.add(TimeRange.WHOLE_DAY);
+        }
+        avaliableTime = listTransfer;
+    }
+
+    // If there are 2 or more events and the number of attendees are more than just one 
+    // check for overlap and return avaliable timerange 
+    else if(events.size() >= 2 && request.getAttendees().size() > 1){
+        if(isOverlapped(eventArray) && eventTakeWholeDay(eventArray) == 0){
+            
+            // Check which event ends the latest since they overlap to find when both 
+            // attendees are free. 
+            int finalEndTime = secondEventEndTime;
+            if(firstEventEndTime > secondEventEndTime){
+                finalEndTime = firstEventEndTime;
+            }
+
+            listTransfer.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, firstEventStartTime, false));
+            listTransfer.add(TimeRange.fromStartEnd(finalEndTime, TimeRange.END_OF_DAY, true));
+        }
+        else{
+            // No overlap means a middle free space.
+            listTransfer.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, firstEventStartTime, false));
+            listTransfer.add(TimeRange.fromStartEnd(firstEventEndTime, secondEventStartTime, false));
+            listTransfer.add(TimeRange.fromStartEnd(secondEventEndTime, TimeRange.END_OF_DAY, true));
+        }
+        avaliableTime = listTransfer;
+    }
+
+    // This else statements mostly applies to meeting requests that only
+    // specifies for one person.
+    else{
+        if(isOverlapped(eventArray)){
+            
+            // If the one person has a meeting thats lasts the whole day
+            // they have no free space.
+            if(eventTakeWholeDay(eventArray) == events.size()){
+                listTransfer.clear();
+            }
+            else{
+                listTransfer.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, firstEventStartTime, false));
+                listTransfer.add(TimeRange.fromStartEnd(secondEventEndTime, TimeRange.END_OF_DAY, true));
+            }
+        }
+        else if(!isOverlapped(eventArray)){
+
+            long freeSpace = ((secondEventStartTime) - (firstEventEndTime));
+
+            // If the request duration is greater than avaliable free time the 
+            // program returns no timerange avaliable.
+            if (request.getDuration() > freeSpace){
+                listTransfer.clear();
+            }
+            // If its not greater than the avaliable free time then return
+            // duration left after event.
+            else{
+                listTransfer.add(TimeRange.fromStartDuration(firstEventEndTime, requestDuration));
+            }
+        }
+        avaliableTime = listTransfer;
+    }
+    return avaliableTime;
+  }
+
+  // Sets times of earliest event and latest events. If event
+  // takes more than a day then ignore.
+  void SetTimes(Event [] eventArray){
```

Comment:
Also I don't understand the purpose of this method - the earliest and latest events are being recorded, but what about other events in the middle that also have space in between? Also doesn't this assume the events are sorted?